### PR TITLE
Feature/1750 migrate app to dp renderer

### PR DESCRIPTION
**Pull request by mr-nick17**
### What

- ✅ [Convert app to use dp-renderer](https://trello.com/c/40dibcN7/1750-migrate-dp-frontend-filter-dataset-controller-to-use-dp-renderer)
- Enabled linter and implemented suggestions

### How to review

- Sense check
- All CI tasks pass
- Templates/models are lift and shift from `dp-frontend-renderer` and `dp-frontend-models`

Have a go locally
- Pull this branch `make debug`
- Port forward onto sandbox web api router
- Run `sixteens`
- Check a cmd filter record

### Who can review

!me

### CODE REVIEW

**Review by lindenmckenzie — COMMENTED** [review]
I've left some comments - I'd also like to run the service, have you got a go-to cmd filter record journey I could look at once I've connected to sandbox?

File: README.md [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
+| HEALTHCHECK_CRITICAL_TIMEOUT | 90s                         | The time taken for the health changes from warning state to critical due to subsystem check failures |
+| HEALTHCHECK_INTERVAL         | 30s                         | The time between calling healthcheck endpoints for check subsystems                                  |
+| MAX_DATASET_OPTIONS          | 200                         | maximum number of IDs that will be requested to dataset API in a single call as query parmeters      |
+| PATTERN_LIBRARY_ASSETS_PATH  | ""                          | Pattern library location                                                                             |
+| PPROF_TOKEN                  | ""                          | The profiling token to access service profiling                                                      |
+| SEARCH_API_AUTH_TOKEN        | n/a                         | The token used to access the Search API                                                              |
+| SITE_DOMAIN                  | string                      | Domain taken from environment configs                                                                |
```
[the remark this answers is not in the document]
Comment (mr-nick17) [reply README.md:27]: I've added the relevant ones already https://github.com/ONSdigital/dp-configs/pull/1329 and will remove the renderer references once the app is in prod (will create a ticket)

File: config/config.go [hunk 2]
```diff
@@ -38,15 +54,16 @@ func Get() (cfg *Config, err error) {
 		APIRouterURL:               "http://localhost:23200/v1",
 		BatchMaxWorkers:            100,
 		BatchSizeLimit:             1000,
-		BindAddr:                   ":20001",
+		BindAddr:                   "localhost:20001",
```
[the remark this answers is not in the document]
Comment (mr-nick17) [reply config/config.go:57]: Depends on where you look 👀; [frontend dataset controller](https://github.com/ONSdigital/dp-frontend-dataset-controller/blob/d94900f379cb72e2ee376a87dd09281c86c651b8/config/config.go#L54) and [frontend filter flex](https://github.com/ONSdigital/dp-frontend-filter-flex-dataset/blob/2abf53ef6d60bc6f246f6e66164e8a69d58458dc/config/config.go#L50C3-L50C3). It removes the dialog box you get when you start the app locally

[the remark this answers is not in the document]
Comment (lindenmckenzie) [reply config/config.go:57]: Interesting.

File: mapper/mapper.go [hunk 3]
```diff
@@ -16,20 +16,16 @@ import (
...
-	timeModel "github.com/ONSdigital/dp-frontend-models/model/dataset-filter/time"
+	"github.com/ONSdigital/dp-frontend-filter-dataset-controller/helpers"
+
+	"github.com/ONSdigital/dp-frontend-filter-dataset-controller/model"
+	core "github.com/ONSdigital/dp-renderer/v2/model"
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
+const sixteensVersion = "7d2d294"
+
```
[the remark this answers is not in the document]
Comment (mr-nick17) [reply mapper/mapper.go:28]: The [readme](https://github.com/ONSdigital/dp-renderer#using-sixteens-for-older-pages) suggests placing it in the mapper. I guess the vision for it to be there temporarily as the migration away from sixteens continues...🚀

[the remark this answers is not in the document]
Comment (lindenmckenzie) [reply mapper/mapper.go:28]: Ok, no problem.

File: dates/dates.go [hunk 4]
```diff
@@ -52,7 +52,7 @@ func Sort(dates []time.Time) TimeSlice {
 // ConvertToCoded takes a list of time.Time and converts to a list of strings in
 // the format yyyy.mm
 func ConvertToCoded(dates []time.Time) []string {
```
[the remark this answers is not in the document]
Comment (mr-nick17) [reply dates/dates.go:54]: hmm the linter just suggested preallocating the variable but not to a set array length

**Comment by mr-nick17** [comment]
> 

Try this filter id := `b6e63a4a-32ec-40e4-a5eb-27d70ace95c6`
You should be able to create any filter record (in the env you're working in) and see the changes in the app if you're port forwarding correctly ☮️

File: dates/dates.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (lindenmckenzie) [reply dates/dates.go:54]: hmm - certainly my understanding was that if you know the array length at the start, it's more performant to make the array with the length already. 

https://github.com/alexkohler/prealloc <-- some "interesting" docs here about it. Might be doesn't meet your use case.

[the remark this answers is not in the document]
Comment (mr-nick17) [reply dates/dates.go:54]: Should have read that readme 😊

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**